src/backup.rs

```diff
@@ -78,6 +78,60 @@ impl BackupEngine {
             Ok(())
         }
     }
+    /// Restore from the latest backup
+    /// db_dir:  input db directory
+    /// wal_dir:  input wal directory
+    /// opts: restore options
+    /// example:
+    /// use rocksdb::backup::{BackupEngine, BackupEngineOptions};
+    /// let backup_opts = BackupEngineOptions::default();
+    /// let backup_engine = BackupEngine::open(&backup_opts, &backup_path).unwrap();
+    /// let mut restore_option = rocksdb::backup::RestoreOptions::default();
+    /// restore_option.set_keep_log_files(true); /// true to keep log files
+    /// if let Err(e) = backup_engine.restore_from_latest_backup(&db_path, &wal_dir, &restore_option) {
+    ///     error!("Failed to restore from the backup. Error:{:?}", e);
+    ///     return Err(e.to_string());
+    ///  }
+    pub fn restore_from_latest_backup<D: AsRef<Path>, W: AsRef<Path>>(
+        &mut self,
+        db_dir: D,
+        wal_dir: W,
+        opts: &RestoreOptions,
+    ) -> Result<(), Error> {
+        let db_dir = db_dir.as_ref();
+        let c_db_dir = match CString::new(db_dir.to_string_lossy().as_bytes()) {
+            Ok(c) => c,
+            Err(_) => {
+                return Err(Error::new(
+                    "Failed to convert db_dir to CString \
+                     when restoring from latest backup"
+                        .to_owned(),
+                ));
+            }
+        };
+
+        let wal_dir = wal_dir.as_ref();
+        let c_wal_dir = match CString::new(wal_dir.to_string_lossy().as_bytes()) {
+            Ok(c) => c,
+            Err(_) => {
+                return Err(Error::new(
+                    "Failed to convert wal_dir to CString \
+                     when restoring from latest backup"
+                        .to_owned(),
+                ));
+            }
+        };
+
+        unsafe {
+            ffi_try!(ffi::rocksdb_backup_engine_restore_db_from_latest_backup(
+                self.inner,
+                c_db_dir.as_ptr(),
+                c_wal_dir.as_ptr(),
+                opts.inner,
+            ));
+        }
+        Ok(())
+    }
 }
```

Comment:
@rohitjoshi I'd like to see a test for this functionality. What do you think ?

Reply:
Let me see how can I add the test cases but I have used and validated at https://github.com/rohitjoshi/bhatho/blob/master/src/db/rocks_db.rs

Reply:
Test could be like this I suppose:
1. Create database. 
2. Put into it one key, value. 
3. Create backup. 
4. Put one more key, value.
5. Close database. 
6. Restore database.
7. Inserted key, value on step 4 shouldn't exist in restored database. 

If I understand correctly purpose of the functionality.

Reply:
I will give a try this weekend.